### PR TITLE
feat: Update Syntax Highlighting in Monaco-Kusto to Preserve Coloring on Copy-Paste

### PR DESCRIPTION
**Pull request by morgilad**
# PR: Replace Syntax Highlighting Implementation in Monaco-Kusto

In this PR, I replaced the implementation of syntax highlighting in Monaco-Kusto. The previous approach used a hack that injected CSS for each KQL classified block, rather than following the recommended Monaco Editor method. This hack caused issues with the 'copy-to-clipboard' functionality, preventing it from retaining the syntax highlighting.

## Changes Made
The work is divided into two parts:

### 1. Syntax Highlighting using Monarch Tokens
- The Monarch token provider is now used to handle syntax highlighting efficiently. This approach leverages the built-in capabilities of the Monaco Editor, ensuring quick and accurate tokenization of KQL syntax. The Monarch tokens help in defining lexical rules and assigning appropriate styling to different code elements without resorting to custom CSS injections.
- **Note**: This part of the PR is still a work in progress as the Monarch tokens are not finished. I need to discuss with Matt to understand the best way to define them.

### 2. Semantic Syntax Highlighting
- A semantic token provider is registered to enhance syntax highlighting by using the Kusto Language Service. This method provides context-aware highlighting, making it more precise and relevant to the code's actual semantics. The Kusto Language Service helps in understanding the deeper context of the code, allowing for more accurate and meaningful syntax highlighting beyond basic lexical analysis.

### 3. New public API > getRangeHtml  
Public API for retrieving HTML content with syntax highlighting for clipboard data.

## Additional Changes
- I have upgraded the major part of the package version to 12.0.0 as this is a breaking change.
- I have also made some adjustments to the local development environment and schema.

### CODE REVIEW

File: package/src/extendedGlobalApi/getRangeHtml.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import { editor, IRange } from 'monaco-editor/esm/vs/editor/editor.api';
+
+export function getRangeHtml(model: editor.IModel, range: IRange): string {
+    const { startLineNumber, endLineNumber, endColumn } = range;
+    const isLastLineEmpty = endColumn === 0;
+    const actualLastLine = isLastLineEmpty ? endLineNumber - 1 : endLineNumber;
+    const totalLines = endLineNumber - actualLastLine + 1;
+
+    const lines = new Array(totalLines);
+    for (let lineNumber = startLineNumber, index = 0; lineNumber <= actualLastLine; lineNumber++, index++) {
```
Comment (morgilad) [inline package/src/extendedGlobalApi/getRangeHtml.ts:10]: replace with map

File: package/src/syntaxHighlighting/themes.ts [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        { token: Token.Identifier, foreground: colors.black },
+        { token: Token.ClientParameter, foreground: colors.tealBlue },
+        { token: Token.QueryParameter, foreground: colors.tealBlue },
+        { token: Token.ScalarParameter, foreground: colors.blue },
+        { token: Token.MathOperator, foreground: colors.black },
+        { token: Token.QueryOperator, foreground: colors.orangeRed },
+        { token: Token.Command, foreground: colors.blue },
+        { token: Token.Keyword, foreground: colors.blue },
+        { token: Token.MaterializedView, foreground: colors.darkOrchid },
+        { token: Token.SchemaMember, foreground: colors.black }, // check this one
```
Comment (morgilad) [inline package/src/syntaxHighlighting/themes.ts:65]: remove this comment

File: package/src/syntaxHighlighting/SemanticTokensProvider.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+export class SemanticTokensProvider implements monaco.languages.DocumentSemanticTokensProvider {
+    private readonly classificationsGetter: ClassificationsGetter;
+
+    constructor(classificationsGetter: ClassificationsGetter) {
+        this.classificationsGetter = classificationsGetter;
+    }
+
+    getLegend() {
+        return { tokenTypes: tokenTypes, tokenModifiers: [] };
```
Comment (morgilad) [inline package/src/syntaxHighlighting/SemanticTokensProvider.ts:15]: no need for tokenTypes: tokenTypes

File: package/src/syntaxHighlighting/SemanticTokensProvider.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    }
+
+    async provideDocumentSemanticTokens(model: editor.ITextModel) {
+        const resource = model.uri;
+        const classifications = await this.classificationsGetter(resource);
+        const semanticTokens = classifications.map((classification, index) => {
+            const last = classifications[index - 1];
```
Comment (morgilad) [inline package/src/syntaxHighlighting/SemanticTokensProvider.ts:22]: previous

Comment (morgilad) [reply package/src/syntaxHighlighting/SemanticTokensProvider.ts:22]: previousClassification

**Review by sagivf — APPROVED** [review]

**Pull request merged**